plugins/pci-psp/fu-pci-psp-device.c

```diff
@@ -13,13 +13,46 @@
 
 #include "fu-pci-psp-device.h"
 
+#define FU_CPU_AGESA_SMBIOS_TYPE   40
+#define FU_CPU_AGESA_SMBIOS_LENGTH 0x0E
+#define FU_CPU_AGESA_SMBIOS_OFFSET 4
+
 struct _FuPciPspDevice {
 	FuUdevDevice parent_instance;
 	gboolean supported;
 };
 
 G_DEFINE_TYPE(FuPciPspDevice, fu_pci_psp_device, FU_TYPE_UDEV_DEVICE)
 
+static void
+fu_pci_psp_device_get_agesa_version(FuDevice *device)
+{
+	FuContext *ctx = fu_device_get_context(device);
+	g_autoptr(GError) error_local = NULL;
+	const gchar *agesa_stream;
+	g_auto(GStrv) split = NULL;
+	g_autofree gchar *summary = NULL;
+
+	/* get the AGESA stream e.g. `AGESA!V9 StrixKrackanPI-FP8 1.1.0.0a` */
+	agesa_stream = fu_context_get_smbios_string(ctx,
+						    FU_CPU_AGESA_SMBIOS_TYPE,
+						    FU_CPU_AGESA_SMBIOS_LENGTH,
+						    FU_CPU_AGESA_SMBIOS_OFFSET,
+						    &error_local);
```

Comment:
I don't think this is going to be emulated -- I think we probably need something `fu_device_get_smbios_string()` which can record and replay the emulation data. I can do this on Monday if that helps.

Reply:
Actually; I had 5 minutes -- can you cherry-pick f97c841ff6f328da9e6c669f7e205bc383ef9f6c and use that API instead -- you'll have to rerecord the emulation data but then it might work okay.